link_linux.go

```diff
@@ -951,14 +952,16 @@ func (h *Handle) linkModify(link Link, flags int) error {
 	// TODO: support extra data for macvlan
 	base := link.Attrs()
 
-	if base.Name == "" {
+	// if tuntap, then the name can be empty, OS will provide a name
+	_, isTuntap := link.(*Tuntap)
+
+	if base.Name == "" && !isTuntap {
 		return fmt.Errorf("LinkAttrs.Name cannot be empty!")
 	}
 
 	if tuntap, ok := link.(*Tuntap); ok {
```

Comment:
this could be optimized by assigning the tuntap in line 956 already and having the condition in line 958 read 
```
if isTuntap {
}
```

Reply:
don't see why not

Reply:
you will need to change line 956 to get the object itself